lib/_http_server.js

```diff
@@ -344,7 +344,9 @@ function connectionListenerInternal(server, socket) {
   socket.on('timeout', socketOnTimeout);
 
   var parser = parsers.alloc();
-  parser.reinitialize(HTTPParser.REQUEST, parser[is_reused_symbol]);
+  parser.reinitialize(HTTPParser.REQUEST,
+                      parser[is_reused_symbol],
+                      server[kIncomingMessage]);
```

Comment:
Isn't this reusing the `IncomingMessage` constructor function instead of the concrete instance? If I understand the code correct the `IncomingMessage` instance is created in `parserOnHeadersComplete`.
If I'm correct I wonder why the dedicated `test-httparser-reuse.js` is not detecting this.

Reply:
@Drieger This one is still open and if I'm correct this is a no go.

Reply:
You are right @Flarna I'll investigate this more in depth.

Reply:
@Flarna `kIncomingMessage` is really a function as you said. I fixed the test to catch this. 
I  will also create a new `object` to be the resource for the server. Some other changes will be necessary as well. I will update the PR as soon as I this is working.